### ISSUE TITLE
"Smart by Bond" devices don't work with new plugin version

### ISSUE THREAD

**Issue by FlyingDiver** [FlyingDiver/Indigo-BondHome#5]
"FYI, the latest plugin update broke my device and bridge. I tried opening and saving the bridge and device and when that didn't work I tried creating as new but nothing worked. I believe it may not have my device as an option (I have one of those fans with the integrated "smart by bond" bridge). I reverted to the previous plugin and everything works now, but I wanted to let you know."

**Comment by FlyingDiver** [comment]
New Plugin:
```
Starting plugin "Bond Home 0.1.2" (pid 36521)
   Bond Home Debug                 logLevel = 5
   Started plugin "Bond Home 0.1.2"
   Bond Home                       Starting BondHome
   Bond Home                       Office Fan Bond: Starting bondBridge Device 798899409
   Bond Home Debug                 BondHome __init__ address = 192.168.1.100, token = *********
   Bond Home Debug                 get_bridge_version()
   Bond Home Debug                 Office Fan Bond: BondHome version: {u'uptime_s': 19226, u'branding_profile': u'MINKA_F747L', u'target': u'breck-spitfire', u'make': u'Minka', u'fw_date': u'Thu Oct 22 18:39:59 UTC 2020', u'fw_ver': u'v2.15', u'api': 2, u'model': u'F747L', u'bondid': u'KSMWTCE82430', u'_': u'9664b0f7', u'upgrade_http': True}
   Bond Home Debug                 get_bridge_info()
   Bond Home Error                 exception in deviceStartComm(Office Fan Bond): 404 Client Error: Not Found for url: http://192.168.1.100/v2/bridge

Mar 29, 2021 at 7:19:42 PM
   Bond Home                       Office Fan through plugin: Starting bondDevice Device 484200233
   Bond Home Error                 Error in plugin execution ServerReplacedElem:

Traceback (most recent call last):
  File "/Library/Application Support/Perceptive Automation/Indigo 7.5/IndigoPluginHost.app/Contents/Resources/PlugIns/plugin_base.py", line 1179, in deviceUpdated
  File "plugin.py", line 134, in deviceStartComm
KeyError: (u'798899409',)
```

**Comment by FlyingDiver** [comment]
Bridge:
```
   Bond Home                       Bond Bridge: Starting bondBridge Device 41423886
   Bond Home Debug                 BondHome __init__ address = 192.168.111.124, token = 1baf3aa2a88bea35
   Bond Home Debug                 get_bridge_version()
   Bond Home Debug                 Bond Bridge: BondHome version: {u'uptime_s': 1237299, u'branding_profile': u'O_BD-1000', u'target': u'zermatt', u'boot_patch': u'not outdated', u'make': u'Olibra', u'fw_date': u'Wed Nov 11 15:50:17 UTC 2020', u'fw_ver': u'v2.15.8', u'api': 2, u'model': u'BD-1000', u'bondid': u'ZZBL41370', u'_': u'6e34650f', u'upgrade_http': True}
   Bond Home Debug                 get_bridge_info()
   Bond Home Debug                 Bond Bridge: BondHome info: {u'name': u'Great Room Bridge', u'frequencies': [[38, 38], [300000, 321500], [336000, 399500], [410500, 450000]], u'location': u'Great Room', u'bluelight': 255, u'compliance_region': u'us', u'_': u'9db946c8'}
```

**Comment by FlyingDiver** [comment]
Hmm.  Need to add a separate device type for the Smart by Bond, or change the device definition on the fly.

**Comment by FlyingDiver** [comment]
```
Starting plugin "Bond Home 0.0.2" (pid 72173)
   Bond Home Debug                 logLevel = 5
   Started plugin "Bond Home 0.0.2"
   Bond Home                       Starting BondHome
   Bond Home                       Office Fan Bond: Starting bondBridge Device 798899409
   Bond Home Debug                 {
    "798899409": {
        "1": {
            "_": "a49f7178",
            "actions": [
                "TurnOn",
                "TurnOff",
                "TogglePower",
                "SetSpeed",
                "IncreaseSpeed",
                "DecreaseSpeed",
                "TurnLightOn",
                "TurnLightOff",
                "ToggleLight",
                "SetBrightness",
                "IncreaseBrightness",
                "DecreaseBrightness",
                "CycleBrightness",
                "SetTimer",
                "BreezeOn",
                "BreezeOff",
                "SetBreeze",
                "SetDirection",
                "ToggleDirection"
            ],
            "addr": {
                "_": "ecf672cb"
            },
            "location": "Office",
            "name": "Office Fan",
            "properties": {
                "_": "dac92e8d"
            },
            "skeds": {
                "_": "68bf209d"
            },
            "state": {
                "_": "465572dd"
            },
            "type": "CF"
        }
    }
}
   Bond Home                       Office Fan Change Direction: Starting bondRelay Device 516262
   Bond Home                       Office Fan Light Brightness: Starting bondRelay Device 1621741078
   Bond Home                       Office Fan Light On/Off: Starting bondRelay Device 273526006
   Bond Home                       Office Fan On/Off: Starting bondRelay Device 18599189
   Bond Home                       Office Fan through plugin: Starting bondDevice Device 484200233
```

**Comment by FlyingDiver** [comment]
https://forums.indigodomo.com/viewtopic.php?f=216&t=25095

**Comment by FlyingDiver** [comment]
https://github.com/FlyingDiver/Indigo-BondHome/releases/tag/0.2.0

**Comment by FlyingDiver** [comment]
Fixed in 0.2.0, I think.

**Comment by Koreysherwin** [comment]
New  is found By plug in, but cannot select device in actions:

<img width="389" alt="Screen Shot 2021-08-09 at 2 00 35 PM" src="https://user-images.githubusercontent.com/51178437/128773837-c4d845db-6dc8-4643-b292-7359d90c2adf.png">
<img width="607" alt="Screen Shot 2021-08-09 at 2 01 22 PM" src="https://user-images.githubusercontent.com/51178437/128773920-d0ec4eb5-a241-4406-bf67-68446f0f4157.png">


`   Bond Home                       
{
    "KSMWWDH99949": {
        "1": {
            "_": "8b67807a",
            "actions": [
                "TurnOn",
                "TurnOff",
                "TogglePower",
                "SetSpeed",
                "IncreaseSpeed",
                "DecreaseSpeed",
                "TurnLightOn",
                "TurnLightOff",
                "ToggleLight",
                "SetBrightness",
                "IncreaseBrightness",
                "DecreaseBrightness",
                "CycleBrightness",
                "SetTimer",
                "BreezeOn",
                "BreezeOff",
                "SetBreeze",
                "SetDirection",
                "ToggleDirection"
            ],
            "addr": {
                "_": "fadc0c3c"
            },
            "location": "Family Room",
            "name": "Fan",
            "properties": {
                "_": "937f1ea1"
            },
            "skeds": {
                "_": "798c6751"
            },
            "state": {
                "_": "5b12bbda"
            },
            "type": "CF"
        }
    },
    "ZZDE62994": {
        "0ef16118": {
            "_": "08cfd78f",
            "actions": [
                "ToggleDirection",
                "SetSpeed",
                "StartDimmer",
                "ToggleLight",
                "TurnOff",
                "Stop",
                "TogglePower",
                "TurnOn",
                "IncreaseSpeed",
                "DecreaseSpeed",
                "SetDirection",
                "TurnLightOn",
                "TurnLightOff"
            ],
            "commands": {
                "_": "f1809285"
            },
            "location": "Family Room",
            "name": "fan remote",
            "properties": {
                "_": "b1f6f625"
            },
            "skeds": {
                "_": "b846bdc2"
            },
            "state": {
                "_": "afc03cd4"
            },
            "template": "A3a",
            "type": "CF"
        }
    }
}`

**Comment by FlyingDiver** [comment]
So you can create it OK, but no direct control (Indigo UI) or via actions.  Hmm.  OK, I'll look at it.

**Comment by Koreysherwin** [comment]
Let me know if I can be of assistance..

**Comment by KCFT23** [comment]
I have been using a very old version of the plugin because it worked with my Smart by Bond fan.  However, now that Indigo will not support python 2 plugins, I am back trying to make it work.  I have been testing and it seems the issue is in the discovery of the device.  If I choose "Bond Device" (I know I should not, but this is for testing) when I create a new device, it does find my fan in "Discovered Devices".  I can create the device but it doesn't work.  The error is:

   Bond Home Error                 exception in device_start_comm(Office Fan Bond): 'bluelight'

I believe that is why you created a new "Smart by Bond Internal Bridge" option when creating a new device.  However, if I try to use that option then my fan does not appear in the "Discovered Devices".  I believe if you are close to make it work, let me know if I can help with more troubleshooting.  Thanks

**Comment by KCFT23** [comment]
This is what I get when creating a new device choosing Smart by Bond:
Starting plugin "Bond Home 2022.0.4" (pid 23229)
   Bond Home Debug                 logLevel = 5
   Started plugin "Bond Home 2022.0.4"
   Bond Home                       Starting BondHome
   Bond Home                       new device: Starting smartBond Device XXX
   Bond Home Debug                 BondHome __init__ address = 192.168.1.100, token = XXX
   Bond Home Debug                 get_bridge_version
   Bond Home Debug                 new device: BondHome version: {'target': 'breck-spitfire', 'fw_ver': 'v3.5', 'fw_date': 'Thu Nov  3 13:51:09 UTC 2022', 'uptime_s': 71, 'make': 'Minka', 'model': 'F747L', 'branding_profile': 'MINKA_F747L', 'bondid': 'KSMWTCE82430', 'upgrade_http': True, 'api': 2, '_': '7094cc81', '__': '7094cc81'}
   Bond Home Debug                 get_device_list
   Bond Home Debug                 Service KSMWTCE82430._bond._tcp.local. of type _bond._tcp.local. state changed: ServiceStateChange.Added
   Bond Home Debug                 Found BondBridges: {'KSMWTCE82430._bond._tcp.local.': 'KSMWTCE82430.local.'}
   Bond Home Debug                 get_device_list: {'_': 'c75e25f1', '__': '00000000', '1': {'_': '86e7356c'}}
   Bond Home Debug                 get_device: 1
   Bond Home Debug                 new device: known_devices:
{'KSMWTCE82430': {'1': {'name': 'Office Fan', 'location': 'Office', 'type': 'CF', 'actions': ['TurnOn', 'TurnOff', 'TogglePower', 'SetSpeed', 'IncreaseSpeed', 'DecreaseSpeed', 'TurnLightOn', 'TurnLightOff', 'ToggleLight', 'SetBrightness', 'IncreaseBrightness', 'DecreaseBrightness', 'CycleBrightness', 'SetTimer', 'BreezeOn', 'BreezeOff', 'SetBreeze', 'SetDirection', 'ToggleDirection'], '_': '86e7356c', '__': '333137b1', 'state': {'_': '3a49b834'}, 'properties': {'_': '3569eb3a'}, 'skeds': {'_': '00000000'}, 'power_cycle_state': {'_': '10f3bd91'}, 'addr': {'_': '915d63f7'}}}}
   Bond Home Debug                 udp_start() socket listener started

However, there are two symptoms of a problem:

1) Nothing appears under Discovered Bridges 
<img width="478" alt="image" src="https://user-images.githubusercontent.com/56086225/224744342-36585cd6-37dc-4d04-b643-bcb261580c9b.png">

2) When trying to create a device, nothing appears under Bridge
<img width="479" alt="image" src="https://user-images.githubusercontent.com/56086225/224744900-6077f6ca-7710-4228-a347-8889c95dfaa1.png">

**Comment by KCFT23** [comment]
Alright, I made some progress.  I guess I didn't understand the instructions completely.  So, I created a "Smart by Bond Internal Bridge" device with the IP address and local token (discovered bridges was none as shown in screenshot above).  Then I created a "Bond Home Bridge" device with the IP address and local token, but in this case there was a discovered bridge with the name of my fan.  Finally, I started creating devices for fan on/off, light on/off, etc., pointing to the "Bond Home Bridge" device.  

I get an error when loading the plugin:  
Bond Home Error                 exception in device_start_comm(Office Fan through plugin): 'bluelight'

However, the commands to turn light on/off, fan on/off, set fan speed, and change direction work!  The only one that I have tried so far and is not working is increase/decrease brightness.  I get the following error:
   Bond Home Error                 Error in plugin execution ExecuteAction:

  File "plugin.py", line 379, in actionControlDevice
  File "/Library/Application Support/Perceptive Automation/Indigo 2022.2/Plugins/BondHome.indigoPlugin/Contents/Server Plugin/bondhome.py", line 84, in device_action
  File "/Library/Application Support/Perceptive Automation/Indigo 2022.2/IndigoPluginHost3.app/Contents/Resources/PlugIns/requests/models.py", line 960, in raise_for_status
type: 400 Client Error: Bad Request for url: http://192.168.1.100/v2/devices/1/actions/IncreaseBrightness

I am happy though because this works for most of my commands.  Thanks!!!
